_nx_cugraph/__init__.py

```diff
@@ -343,16 +343,21 @@ def update_env_var(varname):
     return d
 
 
-def _check_networkx_version() -> tuple[int, int] | tuple[int, int, int]:
+def _check_networkx_version(nx_version=None) -> tuple[int, int] | tuple[int, int, int]:
     """Check the version of networkx and return ``(major, minor)`` version tuple."""
     import re
     import warnings
 
     import networkx as nx
 
-    version_major, version_minor, *version_bug = nx.__version__.split(".")[:3]
+    if nx_version is None:
+        nx_version = nx.__version__
+    version_major, version_minor, *version_bug = nx_version.split(".")[:3]
```

Comment:
Instead of parsing manually, consider using `packaging.version.Version`.

Reply:
We had been using `packaging.version.parse` in tests, but this was removed in https://github.com/rapidsai/cugraph/pull/4629 (a very large PR), and other discussions are in https://github.com/rapidsai/cugraph/pull/4571.

What I like about doing this manually is that we have full control and can add functionality and tests whenever there is a regression, b/c although networkx versions are very predictable, sometimes dev versions can be formatted oddly. I expect this to be low-maintenance, and that we will catch issues very promptly.

https://github.com/rapidsai/nx-cugraph/pull/22 will also help us use the version tuple correctly.